types/yup/index.d.ts

```diff
@@ -202,12 +248,17 @@ export interface TestOptions {
     /**
      * Test function, determines schema validity
      */
-    test: (this: TestContext, value: any) => boolean | ValidationError | Promise<boolean | ValidationError>;
+    test: (
+        this: TestContext,
+        value: any
+    ) => boolean | ValidationError | Promise<boolean | ValidationError>;
 
     /**
      * The validation error message
      */
-    message?: string;
+    message?:
+        | string
+        | ((params: object & Partial<TestMessageParams>) => string);
```

Comment:
Since test has both an config object style callsign and an arguments style callback, I also added the new changes to `message` here too.